sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs

```diff
@@ -78,7 +80,66 @@ private string GetSessionFilePath()
             string className = testAdapter.ClassName.Substring(testAdapter.ClassName.LastIndexOf('.') + 1);
 
             string fileName = name + (IsAsync ? "Async" : string.Empty) + ".json";
-            return Path.Combine(TestContext.CurrentContext.TestDirectory,
+
+            var path = TestContext.CurrentContext.TestDirectory;
```

Comment:
This is an enhancement that we would want to enable for all Track 2 tests. It will be really helpful for users.

It might be better to use an attribute that we include in https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/Directory.Build.props.

```
  <ItemGroup>
    <AssemblyAttribute Include="System.Reflection.AssemblyMetadataAttribute">
    <_Parameter1>SourcePath</_Parameter1>
    <_Parameter2>$(MSBuildProjectDirectory)</_Parameter2>
    </AssemblyAttribute>
  </ItemGroup>
```

We can then retrieve the path like this from within RecordedTestBase:
```cs
string path = ((AssemblyMetadataAttribute) GetType().Assembly.GetCustomAttribute(typeof(AssemblyMetadataAttribute))).Value;
```

Reply:
We were only going to do this for MGMT plane, are you suggesting we do this for data plane as well?

This suggestion seems simpler as well if we can get it from this property.

Reply:
I think making this change both for track 1 and track 2 makes sense.